docusaurus/docs/React/guides/sdk-state-management.mdx

```diff
@@ -0,0 +1,119 @@
+
+The SDK comes with [`ChatContext`](../components/contexts/chat-context.mdx) which holds (among other things) currently active channel and forwards LLC instance passed to the [`Chat`](../components/core-components/chat.mdx) component.
+
+:::note
+You can use either `channel` prop on the `Channel` component or `setActiveChannel` function. You cannot use both at the same time.
+:::
+
+Channel as a prop example:
+
+```tsx
+import { useState, useEffect } from 'react';
+import { Channel, useChatContext } from 'stream-chat-react';
+
+export const ChannelWrapper = ({ channelId, channelType = 'messaging', children }) => {
+  const [activeChannel, setActiveChannel] = useState(undefined);
+  const { client } = useChatContext();
+
+  useEffect(() => {
+    if (!channelId) return;
+
+    const channel = client.channel(channelId, channelType);
+
+    setActiveChannel(channel);
+  }, [channelId, channelType]);
+
+  return <Channel channel={activeChannel}>{children}</Channel>;
+};
+```
+
+Example with `setActiveChannel` function:
+
+```tsx
+import { useEffect } from 'react';
+import { useCreateChatClient, useChatContext, Chat, Channel } from 'stream-chat-react';
+
+const ActiveChannelSetter = ({ channelId, channelType }) => {
+  const { client, setActiveChannel } = useChatContext();
+
+  useEffect(() => {
+    const channel = client.channel(channelId, channelType);
+    setActiveChannel(channel);
+  }, [channelType, channelId]);
+
+  return null;
+};
+
+const App = () => {
+  const client = useCreateChatClient(userData);
+
+  if (!client) return <div>Loading...</div>;
+
+  return (
+    <Chat client={client}>
+      <ActiveChannelSetter channelId='random' channelType='messaging' />
+      <Channel>{/** ...other components... */}</Channel>
+    </Chat>
+  );
+};
+```
+
+Afterwards - to access currently active channel you can use `useChatContext` hook as well from any component (or custom hook) which is either direct or indirect child of a Chat component:
+
+```tsx
+import { useChatContext } from 'stream-chat-react';
+
+export const useActiveChannel = () => {
+  const { channel } = useChatContext();
+
+  return channel;
+};
+```
+
+Currently active [channel state](https://github.com/GetStream/stream-chat-react/blob/master/src/context/ChannelStateContext.tsx#L36-L56) can be accessed through the [`ChannelStateContext`](../components/contexts/channel-state-context.mdx) with the help of the `useChannelStateContext` hook - meaning any component which is either direct or indirect child of the `Channel` component can access such state.
```

Comment:
This is very funny that the same object can be accessed from 2 different contexts :)

Reply:
I agree with @MartinCupela here, let's not even mention the `channel` in the `ChatContext` and stick with `ChannelStateContext` everywhere instead!